aws-django-voting-app/__main__.py

```diff
@@ -0,0 +1,381 @@
+app_ecr_repo = aws.ecr.Repository("app-ecr-repo",
+    image_tag_mutability="MUTABLE")
+
+# Attaching an application life cycle policy to the storage
+app_lifecycle_policy = aws.ecr.LifecyclePolicy("app-lifecycle-policy",
+    repository=app_ecr_repo.name,
+    policy="""{
+        "rules": [
+            {
+                "rulePriority": 10,
+                "description": "Remove untagged images",
+                "selection": {
+                    "tagStatus": "untagged",
+                    "countType": "imageCountMoreThan",
+                    "countNumber": 1
+                },
+                "action": {
+                    "type": "expire"
+                }
+            }
+        ]
+    }""")
+
+# The application's backend and data layer: MySQL
+
+# Creating an RDS instance requires having two subnets
+extra_rds_subnet = aws.ec2.Subnet("extra-rds-subnet",
+    cidr_block="172.31.128.0/20",
+    availability_zone=availability_zone + "b",
+    vpc_id=app_vpc)
+
+# Both subnets are assigned to a SubnetGroup used by the RDS instance
+app_database_subnetgroup = aws.rds.SubnetGroup("app-database-subnetgroup",
+    subnet_ids=[app_vpc_subnet.id, extra_rds_subnet.id])
+
+# An RDS instnace is created to hold our MySQL database
+mysql_rds_server = aws.rds.Instance("mysql-server",
+    engine="mysql",
+    username=sql_admin_name,
+    password=sql_admin_password,
+    instance_class="db.t2.micro",
+    allocated_storage=20,
+    skip_final_snapshot=True,
+    publicly_accessible=True,
+    db_subnet_group_name=app_database_subnetgroup.id,
+    vpc_security_group_ids=[app_security_group.id])
+
+# Creating a Pulumi MySQL provider to allow us to interact with the RDS instance
+mysql_provider = mysql.Provider("mysql-provider",
+    endpoint=mysql_rds_server.endpoint,
+    username=sql_admin_name,
+    password=sql_admin_password)
+
+# Initializing a basic database on the RDS instance
+mysql_database = mysql.Database("mysql-database",
+    name="votes",
+    opts=pulumi.ResourceOptions(provider=mysql_provider))
+
+# Creating a user which will be used to manage MySQL tables 
+mysql_user = mysql.User("mysql-standard-user",
+    user=sql_user_name,
+    host="%", # "%" indicates that the connection is allowed to come from anywhere
+    plaintext_password=sql_user_password,
+    opts=pulumi.ResourceOptions(provider=mysql_provider))
+
+# The user only needs the "SELECT", "UPDATE", "INSERT", and "DELETE" permissions to function
+mysql_access_grant = mysql.Grant("mysql-access-grant",
+    user=mysql_user.user,
+    host=mysql_user.host,
+    database=mysql_database.name,
+    privileges= ["SELECT", "UPDATE", "INSERT", "DELETE"],
+    opts=pulumi.ResourceOptions(provider=mysql_provider))
+
+# The application's frontend: A Django service
+
+# Creating a target group through which the Django frontend receives requests
+django_targetgroup = aws.lb.TargetGroup("django-targetgroup",
+	port=80,
```

Comment:
It looks like there is a mix of tabs and spaces for indentation. Use spaces throughout.